src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/AssemblyContigAlignmentSignatureClassifier.java

```diff
@@ -0,0 +1,171 @@
+                RDDUtils.split(contigsWithOnlyOneBestConfig, AssemblyContigAlignmentSignatureClassifier::hasOnly2Alignments, true);
+
+        // TODO: 10/29/17 multiple alignments decision tree return contigs with only 2 fine-tuned alignments
+        //       the implementation should send back a Tuple2 of {fine-tuned alignments going to be used for analysis, and List(string, ...) of insertion mappings}
+
+        contigsByRawTypes.put(RawTypes.Cpx, hasTwoAlignmentsOrMore._2);
+
+        decisionTreeForContigsWithTwoAlignments(hasTwoAlignmentsOrMore._1, contigsByRawTypes, broadcastSequenceDictionary);
+
+        return contigsByRawTypes;
+    }
+
+    //==================================================================================================================
+
+    // needs this due to Java naive lambda NOT Spark serializable, and we need a serializable callable
+    private static boolean hasOnly2Alignments(final AlignedContig contigWithOnlyOneConfig) {
+        return contigWithOnlyOneConfig.alignmentIntervals.size() == 2;
+    }
+
+    private static void decisionTreeForContigsWithTwoAlignments(final JavaRDD<AlignedContig> contigsWithOnlyOneBestConfigAnd2AI,
+                                                                final EnumMap<RawTypes, JavaRDD<AlignedContig>> contigsByRawTypes,
+                                                                final Broadcast<SAMSequenceDictionary> broadcastSequenceDictionary) {
+        // first remove overlap
+        final JavaRDD<AlignedContig> preprocessedContigs = contigsWithOnlyOneBestConfigAnd2AI.map(
+                contig -> {
+                    final List<AlignmentInterval> deOverlappedAlignments =
+                            ContigAlignmentsModifier.removeOverlap(contig.alignmentIntervals.get(0),
+                                    contig.alignmentIntervals.get(1), broadcastSequenceDictionary.getValue());
+                    return new AlignedContig(contig.contigName, contig.contigSequence, deOverlappedAlignments,
+                            contig.hasEquallyGoodAlnConfigurations);
+                }
+        );
+
+        // split between the case where both alignments has unique ref span or not
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> hasFullyContainedRefSpanOrNot =
+                RDDUtils.split(preprocessedContigs, AssemblyContigAlignmentSignatureClassifier::hasIncompletePictureDueToRefSpanContainment, false);
+        contigsByRawTypes.put(RawTypes.Incomplete, hasFullyContainedRefSpanOrNot._1);
+
+        // split between same chromosome mapping or not
+        final Tuple2<JavaRDD<AlignedContig>, JavaRDD<AlignedContig>> sameChrOrNot =
+                RDDUtils.split(hasFullyContainedRefSpanOrNot._2, AssemblyContigAlignmentSignatureClassifier::isSameChromosomeMapping, false);
+        contigsByRawTypes.put(RawTypes.TandemDupOrMEIBkpt, sameChrOrNot._2);
```

Comment:
You put this set into the map here under the `TandemDupOrMEIBkpt` key, but then a few lines later you override it with a different set -- why? I'd recommend not putting collections into the map until they're complete for clarity.

Reply:
done